pallets/corporate-actions/src/lib.rs

```diff
@@ -425,6 +433,34 @@ decl_module! {
             // Emit event.
             Self::deposit_event(Event::CAInitiated(caa, id, kind, record_date, details, targets, dwt, wt));
         }
+
+        /// Link the given CA `id` to the given `docs`.
+        /// Any previous links for the CA are removed in favor of `docs`.
+        ///
+        /// The workflow here is to add the documents and initiating the CA in any order desired.
+        /// Once both exist, they can now be linked together.
+        ///
+        /// ## Arguments
+        /// - `id` of the CA to associate with `docs`.
+        /// - `docs` to associate with the CA with `id`.
+        ///
+        /// # Errors
+        /// - `UnauthorizedAsAgent` if `origin` is not `ticker`'s sole CAA (owner is not necessarily the CAA).
+        /// - `NoSuchCA` if `id` does not identify an existing CA.
+        /// - `NoSuchDoc` if any of `docs` does not identify an existing document.
+        #[weight = <T as Trait>::WeightInfo::link_ca_doc(docs.len() as u32)]
+        pub fn link_ca_doc(origin, id: CAId, docs: Vec<DocumentId>) {
```

Comment:
Perhaps add a wrapper function that takes in a Document, stores it in the asset module, and links it in this module.